bench/workgen/workgen.cxx

```diff
@@ -137,6 +137,102 @@ static void *thread_workload(void *arg) {
     return (NULL);
 }
 
+static void *thread_idle_table_cycle_worload(void *arg) {
+
+    WorkloadRunnerConnection *runnerConnection = (WorkloadRunnerConnection *) arg;
+    WorkloadRunner *runner = runnerConnection->runner;
+    WT_CONNECTION *connection = runnerConnection->connection;
+
+    try {
+        runner->start_table_idle_cycle(connection);
+    } catch (WorkgenException &wge) {
+        std::cerr << "Exception while create/drop tables." << std::endl;
+    }
+
+    return (NULL);
+}
+
+int WorkloadRunner::check_timing(const char *name, uint64_t start, uint64_t *stop) {
+    uint64_t last_interval;
+    int msg_err;
+    const char *str;
+    WorkloadOptions *options = &_workload->options;
+
+    msg_err = 0;
+
+    workgen_clock(stop);
+
+    last_interval = ns_to_sec(*stop - start);
+
+    if (last_interval > options->max_idle_table_cycle) {
+        if (options->max_idle_table_cycle_fatal) {
+            msg_err = ETIMEDOUT;
+            str = "ERROR";
+        } else {
+            str = "WARNING";
+        }
+        std::cerr << str << ": Cycling idle table failed because " << name << " took " << last_interval << " seconds which is longer than configured acceptable maximum of " << options->max_idle_table_cycle << std::endl;
+     }
+     return (msg_err);
+}
+
+int WorkloadRunner::start_table_idle_cycle(WT_CONNECTION *conn) {
+
+    //std::string session_config = _workload->_threads[0].options.session_config;
+    int ret, cycle_count;
+    WT_SESSION *session;
+    WT_CURSOR *cursor;
+    char uri[BUF_SIZE];
+    uint64_t start, stop;
```

Comment:
The order of the variable declaration is not correct. Unfortunately dist/function.py doesnt run on c++ code. From that file , here is the expected ordering:

```
# List of legal types in sort order.
types = [
    'struct',
    'union',
    'enum',
    'DIR',
    'FILE',
    'TEST_',
    'WT_',
    'wt_',
    'DWORD',
    'double',
    'float',
    'intmax_t',
    'intptr_t',
    'clock_t',
    'pid_t',
    'pthread_t',
    'size_t',
    'ssize_t',
    'time_t',
    'uintmax_t',
    'uintptr_t',
    'u_long',
    'long',
    'uint64_t',
    'int64_t',
    'uint32_t',
    'int32_t',
    'uint16_t',
    'int16_t',
    'uint8_t',
    'int8_t',
    'u_int',
    'int',
    'u_char',
    'char',
    'bool',
    'va_list',
    'void '
]
```

So your ordering should be:

```
WT_SESSION *session;
WT_CURSOR *cursor;
uint64_t start, stop;
int ret, cycle_count;
char uri[BUF_SIZE];
```

Reply:
Thanks for sharing the ordering of the declarations. Fixed